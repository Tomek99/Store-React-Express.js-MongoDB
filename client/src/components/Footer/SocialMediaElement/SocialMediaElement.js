import React from "react";
import styles from "./SocialMediaElement.module.scss";

function SocialMediaElement({ item }) {
  const { name, url } = item;

  return (
    <li className={styles.SocialMediaElement}>
      <a href={url}>{name}</a>
    </li>
  );
}

export default SocialMediaElement;
