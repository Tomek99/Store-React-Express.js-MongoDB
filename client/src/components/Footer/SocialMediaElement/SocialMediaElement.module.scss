.SocialMediaElement {
    display: flex;
    gap: 10px;
    font-size: 2.5rem;
    margin-top: 15px;

    a {
        display: flex;
        justify-content: center;
        align-items: center;
        width: 50px;
        height: 50px;
        border-radius: 50%;
        background-color: var(--main-color);
        color: #fff;
        cursor: pointer;

        &:hover svg {
            transition: 0.3s linear;
            transform: rotate(360deg);
        }
    }
}
