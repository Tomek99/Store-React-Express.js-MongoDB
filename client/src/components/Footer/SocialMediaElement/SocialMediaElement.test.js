import { render } from "@testing-library/react";
import SocialMediaElement from "./SocialMediaElement";

test("should render SocialMediaElement component", () => {
  const item = { name: "test", url: "www.facebook.com" };
  const { container } = render(<SocialMediaElement item={item} />);

  expect(container).toMatchSnapshot();
});
