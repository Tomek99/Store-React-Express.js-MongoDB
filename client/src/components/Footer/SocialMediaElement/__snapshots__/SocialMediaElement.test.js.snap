// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SocialMediaElement component 1`] = `
<div>
  <li
    class="SocialMediaElement"
  >
    <a
      href="www.facebook.com"
    >
      test
    </a>
  </li>
</div>
`;
